DTAdmin/TimeTable/TimeTableListViewController.swift

```diff
@@ -27,13 +30,35 @@ class TimeTableListViewController: ParentViewController, UITableViewDataSource,
     
     override func viewWillAppear(_ animated: Bool) {
         super.viewWillAppear(animated)
-        self.startActivity()
-        CommonNetworkManager.shared().timeTable { (timeTableList, error) in
-            self.stopActivity()
-            if let error = error {
-                self.showAllert(title: "Error", message: error.localizedDescription, completionHandler: nil)
-            } else {
-                self.timeTableList = timeTableList ?? [TimeTable]()
+        if let subjectID = self.subjectID {
+            self.startActivity()
+            CommonNetworkManager.shared().timeTableSubject(by: subjectID, completion: { (timeTableList, error) in
+                self.stopActivity()
+                if let error = error {
+                    self.showAllert(title: "Error", message: error.localizedDescription, completionHandler: nil)
+                } else {
+                    self.timeTableList = timeTableList ?? [TimeTable]()
+                }
+            })
+        } else if let groupID = self.groupID {
+            self.startActivity()
+            CommonNetworkManager.shared().timeTable(by: groupID, completion: { (timeTableList, error) in
+                self.stopActivity()
+                if let error = error {
+                    self.showAllert(title: "Error", message: error.localizedDescription, completionHandler: nil)
+                } else {
+                    self.timeTableList = timeTableList ?? [TimeTable]()
+                }
+            })
+        } else {
+            self.startActivity()
+            CommonNetworkManager.shared().timeTable { (timeTableList, error) in
+                self.stopActivity()
+                if let error = error {
+                    self.showAllert(title: "Error", message: error.localizedDescription, completionHandler: nil)
```

Comment:
Please use NSLocalizedString for all user faced messages.

Reply:
fixed